src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -359,14 +358,19 @@ public function initialized($id)
     public function getServiceIds()
     {
         $ids = array();
+        $reversedMethodMap = array_change_key_case(array_flip($this->methodMap), \CASE_LOWER);
         foreach (get_class_methods($this) as $method) {
             if (preg_match('/^get(.+)Service$/', $method, $match)) {
```

Comment:
So, shouldn't we remove this and rely sololy on methodMap?

the full method could be reduced to the following implementation, isn't it?:
```php
public function getServiceIds()
{
    return array_keys($this-services + $this->methodMap);
}
````